cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,401 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	calculationsv1 "github.com/vega-project/ccb-operator/pkg/apis/calculations/v1"
+	"github.com/vega-project/ccb-operator/pkg/client/clientset/versioned/fake"
+	v1 "github.com/vega-project/ccb-operator/pkg/client/clientset/versioned/typed/calculations/v1"
+	"github.com/vega-project/ccb-operator/pkg/util"
+)
+
+func TestCreateCalculation(t *testing.T) {
+	testCases := []struct {
+		id               string
+		jsonStr          []byte
+		calculationCount int
+		expected         []calculationsv1.Calculation
+	}{
+		{
+			id:               "no calculation",
+			calculationCount: 0,
+		},
+		{
+			id:               "one calculation gets created",
+			jsonStr:          []byte(`{"teff": "14390.0", "logG": "4"}`),
+			calculationCount: 1,
+		},
+	}
+
+	fakecs := fake.NewSimpleClientset()
+	var fakeClient v1.VegaV1Interface
+	fakeClient = fakecs.VegaV1()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for _, tc := range testCases {
+		s := server{
+			logger: logrus.WithField("test-name", "create calculations test"),
+			ctx:    ctx,
+			client: fakeClient,
+		}
+
+		var jsonStr = tc.jsonStr
+		req, err := http.NewRequest("POST", "/calculations/create", bytes.NewBuffer(jsonStr))
+		if err != nil {
+			t.Fatal(err)
+		}
+		req.Header.Set("Content-Type", "application/json")
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(s.createCalculation)
+		handler.ServeHTTP(rr, req)
+
+		response := Response{}
+		err = json.Unmarshal(rr.Body.Bytes(), &response)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if status := response.StatusCode; status != http.StatusOK && status != 0 {
+			logrus.Infof("Handler returned wrong status code: got %v want %v\n", status, http.StatusOK)
+		}
+
+		var actualData *calculationsv1.Calculation
+
+		err = json.Unmarshal(rr.Body.Bytes(), &actualData)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if rr.Code == http.StatusOK && response.StatusCode == 0 {
+			logrus.WithFields(logrus.Fields{"calculation": actualData.Name, "teff": actualData.Spec.Teff, "logG": actualData.Spec.LogG}).Info("Created calculation using the api server...")
+		} else {
+			logrus.Info("No calculation was created...")
+		}
+	}
+}
+
+func TestDeleteCalculation(t *testing.T) {
```

Comment:
We need 4 test cases here. 
- One calculation exists, we delete it with no error.
- One calculation exists, we delete a non-existing calculation and we expect an error.
- Multiple calculations exist, we delete one of them with no error.
- Multiple calculations exist, we delete a non-existing calculation and we expect an error.

Reply:
Well, I always get 400 - not found here.